### PR TITLE
[LPT] StridedSlice Transformation

### PR DESCRIPTION
**Pull request by v-Golubev**
issue #44067

Validation:

• Accuracy: developer-services/job/accuracy/975/ (reference: developer-services/job/accuracy/976/)
• Performance: developer-services/job/performance/510/

### CODE REVIEW

File: inference-engine/src/low_precision_transformations/src/strided_slice.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    return true;
+}
+
+bool StridedSliceTransformation::canBeTransformed(const TransformationContext& context, std::shared_ptr<Node> operation) const {
+    if (!is_type<ngraph::opset1::StridedSlice>(operation)) {
+        return false;
+    }
+
+    const auto dequantization = NetworkHelper::getDequantization(operation);
+    return !!dequantization.multiply;
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/strided_slice.cpp:98]: 1. use `StridedSliceTransformation::canBeTransformed(operation)`
2. `return !NetworkHelper::getDequantization(operation).empty();`

File: inference-engine/src/low_precision_transformations/src/transformer.cpp [hunk 2]
```diff
@@ -316,6 +318,7 @@ TypeRelaxedReplacer::TypeRelaxedReplacer() {
     make_matcher_type_relaxed<opset1::GroupConvolution>(this);
     make_matcher_type_relaxed<opset1::PRelu>(this);
     make_matcher_type_relaxed<opset1::Subtract>(this);
+    make_matcher_type_relaxed<opset1::StridedSlice>(this);
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/transformer.cpp:321]: we don't wrap in TypeRelaxed, do we really need it?

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/strided_slice_transformation.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+};
+
+INSTANTIATE_TEST_CASE_P(smoke_LPT, StridedSliceTransformation,
+    ::testing::Combine(
+        ::testing::ValuesIn(netPrecisions),
+        ::testing::Values(ngraph::Shape({ 1, 3, 24, 24 })),
+        ::testing::Values(CommonTestUtils::DEVICE_GPU),
+        ::testing::ValuesIn(trasformationParamValues),
+        ::testing::ValuesIn(params)),
+    StridedSliceTransformation::getTestCaseName);
```
Comment (eshoguli) [inline inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/strided_slice_transformation.cpp:99]: General comment about GPU: GPU compiles OpenCL program not so fast, please keep only one actual test case. For example: one value in trasformationParamValues only.

**Review by eshoguli — COMMENTED** [review]
1. Source code looks good, please fix comments before merge.
2. Where is validation links?

File: inference-engine/src/low_precision_transformations/src/strided_slice.cpp (hunk 1 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/strided_slice.cpp]: Done, thanks!

File: inference-engine/src/low_precision_transformations/src/transformer.cpp (hunk 2 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/transformer.cpp]: My bad. We don't need to update precision in moveDequantizationAfter. Fixed, thanks!

File: inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/strided_slice_transformation.cpp (hunk 3 above)
Comment (v-Golubev) [reply inference-engine/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/strided_slice_transformation.cpp:96]: Thanks, done

**Comment by v-Golubev** [comment]
@eshoguli Validation links have been attached. StridedSlice is supported in ConcatTransformation (added workaround).

**Review by eshoguli — APPROVED** [review]
looks good for me

File: inference-engine/src/low_precision_transformations/include/low_precision/strided_slice.hpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2020 Intel Corporation
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/include/low_precision/strided_slice.hpp:1]: 2021

File: inference-engine/src/low_precision_transformations/src/subgraph.cpp [hunk 5]
```diff
@@ -33,6 +33,11 @@ bool isQuantizationPerChannel(const std::shared_ptr<ngraph::Node>& node) {
             continue;
         }
 
+        //WA to support StridedSlice in ConcatTransformation
+        if (ngraph::is_type<opset1::StridedSlice>(input.get_node())) {
+            continue;
+        }
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/subgraph.cpp:39]: is it really necessary? please, remove

**Review by eshoguli — APPROVED** [review]
please fix and merge

File: inference-engine/src/low_precision_transformations/include/low_precision/strided_slice.hpp (hunk 4 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/include/low_precision/strided_slice.hpp]: Thanks, done

File: inference-engine/src/low_precision_transformations/src/subgraph.cpp (hunk 5 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/subgraph.cpp]: Thanks, done

**Pull request merged**